packages/react-reconciler/src/ReactFiberHydrationContext.js

```diff
@@ -248,7 +248,8 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       );
     }
     hydrationParentFiber = fiber;
-    nextHydratableInstance = getFirstHydratableChild(nextInstance);
+    nextHydratableInstance =
+      fiber.tag !== HostText ? getFirstHydratableChild(nextInstance) : null;
```

Comment:
`null` is what `getFirstHydratableChild` would return given a text instance in current implementation.
That's why the behavior hasn't changed.

https://github.com/facebook/react/blob/8227e54ccf32f47e4c6bf5f30d08f84b8fed455d/packages/react-dom/src/client/ReactDOMHostConfig.js#L422